### DISCUSSION TITLE
[@types/google.maps] with ReactJS "ReferenceError: google is not defined"

### DISCUSSION THREAD

**Discussion by ZachHandley** [DefinitelyTyped/DefinitelyTyped#55567] [Issues with a @types package]
Okay I've been googling now for around an hour or two and I can't find _any_ solution for this.

Basically my typings work with VS Code, and all is jolly and good while writing my code. But after compiling I get that error, `google is not defined`

I tried adding a node reference `/// <reference type="google.maps"/>`, I tried adding a module declaration inside a custom `index.d.ts` (`declare module "google.maps";`), I tried importing `import {} from "google.maps"`, I can't think of anything else to try and it's really bugging me.

Is there a sure-fire way to understand where this error is coming from and how to fix it? Google is clearly defined.

Thanks

[a comment by an automated account is not shown]

**Reply by ZachHandley** [reply]
Got it I think that should do it

**Comment by jpoehnelt** [comment]
Please share your tsconfig and version of TypeScript.

**Comment by ZachHandley** [comment]
```{
  "compilerOptions": {
    "target": "es5",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "typeRoots": ["./src/*.d.ts", "./node_modules/@types/"],
    "allowJs": true,
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "forceConsistentCasingInFileNames": true,
    "noFallthroughCasesInSwitch": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "noEmit": true,
    "jsx": "react-jsx"
  },
  "include": [
    "src/**/*",
    "src/index.d.ts"
  ]
}```

TypeScript version 4.3.5, app created using npx create-react-app --template Typescript, had no issues for a minute but for some reason this keeps coming back to me. Please also know this is my current tsconfig but it was unchanged until I encountered this error. The type definitions are visible while programming.

**Reply by ZachHandley** [reply]
Wasn't trying to put _all_ of the words into that code section but the formatting gets screwy otherwise

**Reply by ZachHandley** [reply]
I would be happy to share the GitHub folder with you, I fixed this issue before but the steps I took before don't seem to be resolving it now

**Comment by jpoehnelt** [comment]
Where you able to resolve this? I missed the GitHub notification on this. sorry :/

**Reply by ZachHandley** [reply]
No I wasn't, it's still unable to be figured out

**Comment by jpoehnelt** [comment]
A `ReferenceError` is going to be a runtime error and not a TypeScript compilation error. I am guessing that some reference is made to window.google.maps before the Google Maps API has been loaded. How are you loading the API?
